### PR TITLE
BQ e2e Tests Scenarios

### PR DESCRIPTION
**Pull request by priyabhatnagar25**
BQ e2e Tests Scenarios

### CODE REVIEW

**Comment by priyabhatnagar25** [comment]
One pager document for the scenarios created: https://docs.google.com/document/d/1HJf8yVHPqCTmNzKX2wpi7ZaR_vFk4J1k-0oyw6d8tLY/edit?resourcekey=0-MAVQXZfdDk8a0E7mWm4kSw

File: src/e2e-test/features/bigquery/source/BigQuerySourceError.feature [hunk 1]
```diff
@@ -27,6 +27,7 @@ Feature: BigQuery source - Validate BigQuery source plugin error scenarios
       | table          | bqIncorrectTableName        |
       | datasetProject | bqIncorrectDatasetProjectId |
 
+  @BigQuery_Source
```
Comment (Aryan-Verma) [inline src/e2e-test/features/bigquery/source/BigQuerySourceError.feature:30]: Why is this change required? We already have this tag for the entire feature.

File: src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BQValidationExistingTables.java [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+   * Validates the actual data in a BigQuery table against the expected data in a JSON file.
+   * @param table    The name of the BigQuery table to retrieve data from.
+   * @param fileName The name of the JSON file containing the expected data.
+   * @return True if the actual data matches the expected data, false otherwise.
+   */
+  public static boolean validateActualDataToExpectedData(String table, String fileName) throws IOException,
+    InterruptedException, URISyntaxException {
+    Map<String, JsonObject> bigQueryMap = new HashMap<>();
+    Map<String, JsonObject> fileMap = new HashMap<>();
+    Path  bqExpectedFilePath = Paths.get(BQValidationExistingTables.class.getResource("/" + fileName).toURI());
```
Comment (Aryan-Verma) [inline src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BQValidationExistingTables.java]: nit :
```suggestion
    Path bqExpectedFilePath = Paths.get(BQValidationExistingTables.class.getResource("/" + fileName).toURI());
```

File: src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java [hunk 3]
```diff
@@ -924,4 +924,46 @@ public static void createSourceBQUpdateTable() throws IOException, InterruptedEx
...
+                                                          "(ID INT64, Name STRING, " + "Price FLOAT64," +
+                                                          "Customer_Exists BOOL ) ");
+    try {
+      io.cdap.e2e.utils.BigQueryClient.getSoleQueryResult("INSERT INTO `" + datasetName + "." + bqSourceTable + "` " +
+                                                            "(ID, Name, Price, Customer_Exists)" +
+                                                            "VALUES" + "(1, 'Raja Sharma', 200.0, true)");
+
+    } catch (NoSuchElementException e) {
+      // Insert query does not return any record.
+      // Iterator on TableResult values in getSoleQueryResult method throws NoSuchElementException
```
Comment (Aryan-Verma) [inline src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java:941]: We should log an error message here for no records found

File: src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java [hunk 4]
```diff
@@ -924,4 +924,46 @@ public static void createSourceBQUpdateTable() throws IOException, InterruptedEx
...
+                                                          "Price FLOAT64, Customer_Exists BOOL ) ");
+
+    try {
+      io.cdap.e2e.utils.BigQueryClient.getSoleQueryResult("INSERT INTO `" + datasetName + "." + bqTargetTable + "` " +
+                                                            "(ID,  Name, Price, Customer_Exists)" +
+                                                            "VALUES" + "(3, 'Rajan Kumar', 100.0, true)");
+
+    } catch (NoSuchElementException e) {
+      // Insert query does not return any record.
+      // Iterator on TableResult values in getSoleQueryResult method throws NoSuchElementException
```
Comment (Aryan-Verma) [inline src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java:963]: Same for the error message logging

File: src/e2e-test/features/bigquery/source/BigQuerySourceError.feature (hunk 1 above)
Comment (Aryan-Verma) [inline src/e2e-test/features/bigquery/source/BigQuerySourceError.feature]: Why is this change required? We already have this tag for the entire feature.

File: src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java (hunk 3 above)
Comment (Aryan-Verma) [inline src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java:959]: We should log an error message here for no records found

File: src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java (hunk 4 above)
Comment (Aryan-Verma) [inline src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java:1026]: Same for the error message logging

File: src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BQValidationExistingTables.java (hunk 2 above)
Comment (priyabhatnagar25) [reply src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BQValidationExistingTables.java]: Removed the extra space.

File: src/e2e-test/features/bigquery/source/BigQuerySourceError.feature (hunk 1 above)
Comment (priyabhatnagar25) [reply src/e2e-test/features/bigquery/source/BigQuerySourceError.feature]: Hi, Actually the build was failing due to this older test so just added this tag. Reverting it to the previous way now.

File: src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java (hunk 3 above)
Comment (priyabhatnagar25) [reply src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java:959]: Added a log error message to check if the records are not inserted

File: src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java (hunk 4 above)
Comment (priyabhatnagar25) [reply src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java:1026]: Added a log error message to check if the records are not inserted

**Review by Aryan-Verma — APPROVED** [review]
Approving this since the github actions failures are independent of these changes and are investigated separately.
